example/90-shoot.yaml

```
@@ -320,7 +320,9 @@ spec:
   #   statusTaints:
   #     - "reservedForTenant"
   #   newPodScaleUpDelay: 0s
-  #   maxEmptyBulkDelete: 10
+  #   maxEmptyBulkDelete: 10 # Deprecated, will be removed in gardener v1.122
```

I don't think we can remove this field so fast. This is a breaking API change for end-users. We should probably forbid using it with the next Kubernetes version  (v1.33) and only remove it once we drop support for that version.

---

```suggestion
  #   maxEmptyBulkDelete: 10 # Deprecated, will be removed once gardener drops support for Kubernetes v1.33
```
That makes sense, does this deprecation notice seem fine then?

Just from an understanding POV, I was under the assumption that usually gardener supports breaking API changes for three versions before dropping, is that not always the case?

---

You could also send a warning when some `Shoot` still uses this field: https://github.com/gardener/gardener/blob/master/pkg/api/core/shoot/warnings.go

> Just from an understanding POV, I was under the assumption that usually gardener supports breaking API changes for three gardener minor versions before dropping, is that not always the case?

Only for APIs that are not exposed to end-users. They typically need more time to adopt them (and don't have control over when new Gardener versions are rolled out). Hence, we usually bundle them together with Kubernetes version upgrades. See https://github.com/gardener/gardener/blob/master/docs/development/process.md#deprecations-and-backwards-compatibility

---

Done the same, thanks for the explanation.